runway/data_types.py

```diff
@@ -212,12 +212,17 @@ def random_sample(result_of_setup, args):
     :type sampling_std: float, optional
     :raises MissingArgumentError: A missing argument error if length is not specified
     """
-    def __init__(self, length=None, name=None, sampling_mean=0, sampling_std=1):
-        if length is None: raise MissingArgumentError('length')
+    def __init__(self, length=None, name=None, default=None, sampling_mean=0, sampling_std=1):
+        if length is None:
+            raise MissingArgumentError('length')
+        if default is not None and len(default) != length:
+            msg = 'default argument does not match expected length'
+            raise InvalidArgumentError(msg)
```

Comment:
If a `default` vector is supplied during construction we could infer the length from that. I'm also fine with making the `length` required nonetheless.

Reply:
Good call!